### PR TITLE
[Bugfix] Fix Breeze-TTS-2 repeated silence and audio EOS handling

### PR DESCRIPTION
**Pull request by linyueqian**
## Purpose

Breeze-TTS-2 can repeat codec tokens for tens of seconds before reaching audio EOS. In a reproduced English request, the baseline spoke only “Hello” and then generated 29.84 seconds of trailing quiet. The port omitted the generated-codebook-0 repetition penalty used by the [official inference example](https://github.com/breezeblue-ai/breeze-tts/blob/008f769016b0a24711becd7a4925030bc93f608c/infer.py).

- Apply the sign-aware repetition penalty to generated codebook-0 history, excluding text/reference conditioning. Default to the upstream value of `1.1` and support speech API overrides through `extra_params.repetition_penalty`.
- Keep the scheduler's selected token consistent with the codec frame already generated. Ignore text-tokenizer EOS `1`, which is a valid audio code, while retaining explicit audio EOS `2051`; use `min_tokens=0` to prevent masking a selected token.
- Match the streaming decoder's expected chunk size to the producer's 8-frame chunks, and retain original batch indices when empty terminal requests are filtered out.
- Add regression coverage and document the current greedy/CFG-1 scope relative to the official examples.

The scheduler logits now represent a deterministic selected token, so their sampled-token logprobs are zero. This change does not add stochastic sampling or CFG support, and longer internal pauses and eager streaming performance remain limitations.

## Test Plan

**vLLM Version:** 0.29.0.

**vLLM-Omni Commit:** PR based on `d4dba0ddb`; GPU before/after diagnostics used the same Breeze changes on the earlier `6040bb97e` base plus an unrelated output-accumulation change. Final CPU checks are rerun on the PR branch.

**Environment:** one NVIDIA H100 80 GB for synthesis; PyTorch 2.13.0+cu132 and Transformers 5.14.1. Public `BreezeBlue/Breeze-TTS-2` weights, speaker S0, greedy decoding, CFG 1, default async-chunk deployment, 384-frame output cap. Audio measurements use 20 ms RMS windows below −45 dBFS on untrimmed generated WAVs.

Serve the baseline or this branch:

```bash
vllm serve BreezeBlue/Breeze-TTS-2 --omni \
    --deploy-config vllm_omni/deploy/breeze_tts_2.yaml --port 8091
```

Send the reproduction request:

```bash
curl --fail http://127.0.0.1:8091/v1/audio/speech \
    -H 'Content-Type: application/json' \
    -d '{"model":"BreezeBlue/Breeze-TTS-2","input":"Hello, this is Breeze TTS 2 running on vLLM Omni.","voice":"S0","response_format":"wav","max_new_tokens":384}' \
    --output breeze.wav
```

On this branch, add `"extra_params":{"repetition_penalty":1.0}` to the request to disable the penalty for a controlled comparison. Streaming PCM was also checked with `"stream":true,"stream_format":"audio","response_format":"pcm"`.

```bash
CUDA_VISIBLE_DEVICES="" pytest -o addopts="" -q tests/model_executor/models/breeze_tts_2
pre-commit run --from-ref origin/main --to-ref HEAD --show-diff-on-failure
```

## Test Result

- Regression tests failed before their corresponding fixes. Final Breeze CPU suite: **93 passed, 18 warnings in 4.92 s** with CUDA hidden. Warnings concern dependency deprecations and the version fallback in the source-only validation checkout.
- All applicable pre-commit hooks passed, including mypy, test markers, Ruff, Markdown, and import/SPDX checks.
- Six English/Chinese serving requests reached audio EOS after the fix; Whisper large-v3-turbo recovered the intended text in all six. ASR was used for content checks; generated audio was also reviewed by listening.
- English-short trailing quiet decreased from **29.84 s to 0.46 s** and total duration from **30.40 s to 5.68 s**, with the complete sentence spoken. Disabling only the penalty on the patched server reproduced all **380 baseline codec frames exactly**.
- Chinese-short longest internal quiet interval decreased from **3.02 s to 1.18 s**. Some longer requests still contain roughly 3-second internal pauses.
- Decoding identical saved codec frames with the old and corrected codec settings preserved the quiet intervals, isolating the long-tail generation failure from waveform padding. Real decoder regression tests cover rolling-cache progression, partial final chunks, and mixed batches containing empty terminal requests.

These are functional and signal-level diagnostics, not an optimized throughput benchmark or a claim of full upstream inference parity.

### CODE REVIEW

**Review by Sy0307 — APPROVED** [review]
Nice fix. Thx.
